controllers/front/ProductController.php

```diff
@@ -102,8 +102,9 @@ public function init()
              * allow showing the product
              * In all the others cases => 404 "Product is no longer available"
              */
-            if (!$this->product->isAssociatedToShop() || !$this->product->active) {
-                if (Tools::getValue('adtoken') == Tools::getAdminToken('AdminProducts'.(int) Tab::getIdFromClassName('AdminProducts').(int) Tools::getValue('id_employee')) && $this->product->isAssociatedToShop()) {
+            $isAssociatedToProduct = (bool) (Tools::getValue('adtoken') == Tools::getAdminToken('AdminProducts'.(int) Tab::getIdFromClassName('AdminProducts').(int) Tools::getValue('id_employee')) && $this->product->isAssociatedToShop());
+            if ((!$this->product->isAssociatedToShop() || !$this->product->active) && !(bool)Tools::getValue('preview', 0)) {
```

Comment:
Cast to bool is redundant when using the negation operator.

Reply:
In order to reduce code duplication, I think it would be better to store this in a variable.
```php
$isPreview = ('1' === Tools::getValue('preview'));
```